File: PSP/MusicAppRest/src/main/java/org/springrest/domain/model/Song.java
package org.springrest.domain.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Data
@Entity
@Table(name = "songs")
public class Song {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "song_id")
    private int songId;
    @Column(name = "name")
    private String songName;
    @Column
    private String artist;
    @ManyToMany(mappedBy = "songs")
    @JsonIgnore
    private List<Playlist> playlist;
}

File: Moviles/MusicAppConRest/app/src/main/java/com/example/musicapprest/domain/model/Song.kt
package com.example.musicapprest.domain.model

import com.google.gson.annotations.SerializedName


data class Song(
    @SerializedName("songId")
    val songId: Int = 0,
    @SerializedName("songName")
    val songName: String = "",
    @SerializedName("artist")
    val artist: String = "",
)